cmd/pressure-vessel/main.ino

```diff
@@ -73,19 +73,44 @@ void flash(int n, int p) {
   }
 }
 
+bool pumpOn = false;
+unsigned long startPumpTime = 0;
+int alarmCount = 0; // Global to track alarm state entries.
+
 // This will put arduino into an alarm state i.e. something
 // went wrong.
-void alarmed(int flashes){
-  // Again force relay pin low, we don't want pump on.
+void alarmed(int flashes) {
+  unsigned long alarmStartTime = millis();
+  alarmCount++; // Increment the alarm state counter.
+  
+  // Log that the system has entered the alarm state.
+  Serial.print("Entering alarm state. Count: ");
+  Serial.println(alarmCount);
+
+  // Ensure pump is off for safety.
   relay(LOW);
+  pumpOn = false;
```

Comment:
We should probably have a ```pump(bool)``` function or something to wrap setting of the relay state and changing the ```pumpOn``` global variable.